backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/GitUtils.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SonarLint Core - Commons
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.commons;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.Repository;
+import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
+
+import static org.sonarsource.sonarlint.core.commons.GitBlameUtils.buildRepository;
+
+public class GitUtils {
+
+  private static final SonarLintLogger LOG = SonarLintLogger.get();
+
+  private GitUtils() {
+    // Utility class
+  }
+
+  /**
+   *
+   *
+   */
+  public static List<URI> filterOutIgnoredFiles(Path baseDir, List<URI> filePathsToAnalyze) {
+    try {
+      var repo = buildRepository(baseDir);
+      var ignoredFiles = getIgnoredFiles(repo);
```

Comment:
This may be a long operation for the projects with a lot of ignored files like generated code, libs and IDE configs. Maybe try using `FastIgnoreRule` and `IgnoreNode` API of JGit to check specifically the list of files submitted to analysis against `.gitignore` patterns.